File: pykde-bugs/saveProperties.cc
#include <kapplication.h>
#include <kcmdlineargs.h>
#include <kmainwindow.h>
#include <kconfig.h>

class MainWindow : public KMainWindow
{
    Q_OBJECT;

    public:
      MainWindow() {}
      ~MainWindow() {}

      void saveProperties(KConfig *);
};

void MainWindow::saveProperties(KConfig *config)
{
    config->writeEntry("foo", "bar");
}

int
main (int argc, char **argv)
{
    KCmdLineArgs::init(argc, argv, "test", "test", "test", "1.0");
    KApplication app;
    MainWindow *window = new MainWindow;

    app.setTopWidget(window);
    window->show();
    return app.exec();
}

#include "saveProperties_moc.cc"

File: pykde-bugs/saveProperties.py
#! /usr/bin/env python

"""http://www.riverbankcomputing.com/pipermail/pyqt/2007-August/017052.html"""

import sys

import kdeui
import kdecore

def main():
    kdecore.KCmdLineArgs.init(sys.argv, 'test', 'test', 'test', '1.0')
    application = kdecore.KApplication()
    main_window = MainWindow()
    main_window.show()
    application.exec_loop()

class MainWindow(kdeui.KMainWindow):
    def saveProperties(self, config):
        config.writeEntry('foo.py', 'bar.py')

if __name__ == '__main__':
    main()
